README.md

```diff
@@ -0,0 +1,518 @@
+# Doctest: Test interactive Haskell examples
+
+`doctest` is a tool that checks
+[examples](https://www.haskell.org/haddock/doc/html/ch03s08.html#idm140354810775744)
+and
+[properties](https://www.haskell.org/haddock/doc/html/ch03s08.html#idm140354810771856)
+in Haddock comments.
+It is similar in spirit to the [popular Python module with the same name](https://docs.python.org/3/library/doctest.html).
+
+* [Getting started](#getting-started)
+   * [Installation](#installation)
+   * [A basic example](#a-basic-example)
+   * [Running doctest for a Cabal package](#running-doctest-for-a-cabal-package)
+* [Writing examples and properties](#writing-examples-and-properties)
+   * [Example groups](#example-groups)
+      * [A note on performance](#a-note-on-performance)
+   * [Setup code](#setup-code)
+   * [Multi-line input](#multi-line-input)
+   * [Multi-line output](#multi-line-output)
+   * [Matching arbitrary output](#matching-arbitrary-output)
+   * [QuickCheck properties](#quickcheck-properties)
+   * [Hiding examples from Haddock](#hiding-examples-from-haddock)
+   * [Using GHC extensions](#using-ghc-extensions)
+* [Limitations](#limitations)
+* [Doctest in the wild](#doctest-in-the-wild)
+* [Development](#development)
+* [Contributors](#contributors)
+
+
+# Getting started
+
+## Installation
+
+`doctest` is available from
+[Hackage](http://hackage.haskell.org/cgi-bin/hackage-scripts/package/doctest).
+Install it with:
+
+    cabal update && cabal install doctest
```

Comment:
`cabal install` is not really compatible with a multi-GHC environment.  I will just install `doctest` that works with the current GHC.  I would expect that rather one defines `doctest` as a dependency in the `.cabal` file and then it will be installed locally only, for the GHC version that is used to `cabal test` the project.